SteakBot.Core/Services/MemeService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Newtonsoft.Json;
+using SteakBot.Core.Objects;
+using System.Collections.Generic;
+using System.IO;
+using System.Configuration;
+
+namespace SteakBot.Core.Services
+{
+    public class MemeService
+    {
+        private static readonly string MemeCommandsFileName = ConfigurationManager.AppSettings["memeCommandsRelativeFilePath"];
+        private static readonly string MemeCommandsOriginRelativeFilePath = ConfigurationManager.AppSettings["memeCommandsOriginRelativeFilePath"];
+
+        public delegate void ReloadCommands();
+        public event ReloadCommands ReloadCommandsEvent;
+
+        public IList<MemeCommand> MemeCommands { get; set; }
+
+        public MemeService()
+        {
+            LoadCommands();
+        }
+
+        #region Public methods
+
+        public bool SaveCommand(MemeCommand newCmd)
+        {
+            if (!ValidateNewCommand(newCmd))
+            {
+                return false;
+            }
+
+            MemeCommands.Add(newCmd);
+            using (var fileWriter = new StreamWriter(MemeCommandsFileName))
+            {
+                using (var jsonTextWriter = new JsonTextWriter(fileWriter))
+                {
+                    var jsonSerializer = new JsonSerializer();
+                    jsonSerializer.Serialize(jsonTextWriter, MemeCommands);
+                }
+            }
+
+#if DEBUG
+            // Replace original file.
+            using (var fileWriter = new StreamWriter(MemeCommandsOriginRelativeFilePath))
+            {
+                using (var jsonTextWriter = new JsonTextWriter(fileWriter))
+                {
+                    var jsonSerializer = new JsonSerializer();
+                    jsonSerializer.Serialize(jsonTextWriter, MemeCommands);
+                }
+            }
+#endif
+
+            ReloadCommandsEvent?.Invoke();
+
+            return true;
+        }
+
+        public IList<MemeCommand> LoadCommands()
```

Comment:
Return value is never used so you can change to `void` and remove the `return` statement.